internal/ir/xds.go

```diff
@@ -640,6 +640,11 @@ type HeaderSettings struct {
 	// It defaults to false.
 	PreserveXRequestID bool `json:"preserveXRequestID,omitempty" yaml:"preserveXRequestID,omitempty"`
 
+	// GenerateRequestID configures whether envoy will generate the x-request-id header if it does not exist.
+	// Generating a random UUID4 is expensive so in high throughput scenarios where this feature is not desired it can be disabled.
+	// It defaults to true
```

Comment:
I think that we try to avoid default-true boolean flags. However, not sure if there's a good naming option here for a default-false bool.

Reply:
I can adjust this because it's probably not clear. The _Envoy API _ defaults to true for this feature and Envoy Gateway defaults to an empty field here

Reply:
@guydc does the updated description sound reasonable?